tests/test_awslambda/test_lambda.py

```diff
@@ -1027,6 +1027,54 @@ def test_invoke_function_from_sqs():
     assert False, "Test Failed"
 
 
+@mock_logs
+@mock_lambda
+@mock_dynamodb2
+def test_invoke_function_from_dynamodb():
+    logs_conn = boto3.client("logs")
+    dynamodb = boto3.client('dynamodb')
+    table_name = 'table_with_stream'
+    table = dynamodb.create_table(TableName=table_name,
+                                  KeySchema=[{'AttributeName':'id','KeyType':'HASH'}],
+                                  AttributeDefinitions=[{'AttributeName':'id','AttributeType':'S'}],
+                                  StreamSpecification={'StreamEnabled': True,
+                                                       'StreamViewType': 'NEW_AND_OLD_IMAGES'})
+
+    conn = boto3.client('lambda')
+    func = conn.create_function(FunctionName='testFunction', Runtime='python2.7',
+                                Role='test-iam-role',
+                                Handler='lambda_function.lambda_handler',
+                                Code={'ZipFile': get_test_zip_file3()},
+                                Description='test lambda function executed after a DynamoDB table is updated',
+                                Timeout=3, MemorySize=128, Publish=True)
+
+    response = conn.create_event_source_mapping(
+        EventSourceArn=table['TableDescription']['LatestStreamArn'],
+        FunctionName=func['FunctionArn']
+    )
+
+    assert response['EventSourceArn'] == table['TableDescription']['LatestStreamArn']
+    assert response['State'] == 'Enabled'
+
+    dynamodb.put_item(TableName=table_name, Item={'id': { 'S': 'item 1' }})
+    start = time.time()
+    while (time.time() - start) < 30:
+        result = logs_conn.describe_log_streams(logGroupName='/aws/lambda/testFunction')
+        log_streams = result.get('logStreams')
+        if not log_streams:
+            time.sleep(1)
+            continue
+
+        assert len(log_streams) == 1
+        result = logs_conn.get_log_events(logGroupName='/aws/lambda/testFunction', logStreamName=log_streams[0]['logStreamName'])
+        for event in result.get('events'):
+            if event['message'] == 'get_test_zip_file3 success':
+                return
+        time.sleep(1)
```

Comment:
What is the sleeping for? Is there a way to make this test without needing them?

Reply:
Again following the SQS example, in the test just above. I guess the Lambda is invoked async when this is run in server mode, so we'd really have to wait for the Lambda to finish executing/writing logs

Reply:
Ah, okay. That is a bit annoying, but I can understand it and don't have a better suggestion off the top of my head.